### DISCUSSION TITLE
Using dot notation keys without namespaces

### DISCUSSION THREAD

**Discussion by dbrxnds** [amannn/next-intl#148] [Q&A] [answered]
Hi,

We ran into an issue where we use dot notation for translations but with a flat messages json. 

Wondering if there is a way to make next-intl aware that we use a flat object, so dot notation is ignored and just assumes regular key lookup.

**Comment by amannn** [comment, marked as answer]
Hi David,

nice to hear from you! Currently `next-intl` does not provide an option to configure a custom namespace character (see #105).

Any chance you could transform your flat messages file to a nested one? Here's [a simplified transformation script](https://next-intl.dev/docs/usage/messages#messages-other-styles) for inspiration.

If we'd support dot notations in namespaces, a case like this would be ambiguous:
```json
{
  "one.two": "12",
  "one": {"two": "12"}
}
```

That's why I'd like to avoid supporting it.

**Reply by dbrxnds** [reply]
Thanks for your quick reply. We do have the option to get a nested json, but we have existing keys that override those nested fields. 

eg. we have a key `ADMIN` and another key `ADMIN.GENERAL.DELETE_CONFIRMATION`. In the nested JSON it just ends up with the `ADMIN` key.

If there is no way around it, we will just have to end up spending some time converting those keys that break nesting to keys that will work.

**Reply by amannn** [reply]
I see! Maybe you can use a default key for the ones that would overwrite a whole object (e.g. `ADMIN.TITLE`)?

**Reply by dbrxnds** [reply]
Figured that would be the best solution. That's what we ended up doing. Thanks!

**Reply by deammer** [reply]
@amannn sorry to necro this thread but the Code Sandbox you linked isn't available anymore. Would it be possible to share the code in here?

**Reply by amannn** [reply]
Oh yep, just updated [the link above](https://github.com/amannn/next-intl/discussions/148#discussioncomment-4274218)—thanks for the hint!

**Reply by deammer** [reply]
Super appreciated, thank you!
